tsdb/querier.go

```diff
@@ -256,10 +256,23 @@ func PostingsForMatchers(ctx context.Context, ix IndexReader, ms ...*labels.Matc
 			// We already handled the case at the top of the function,
 			// and it is unexpected to get all postings again here.
 			return nil, errors.New("unexpected all postings")
+
```

Comment:
Just a nit, but I'd prefer staying consistent and not adding blank lines between the `switch` branches.

Reply:
I added the lines to visually group the different cases instead of having a continuous chunk of code: the previous two cases handle the `~".*"` matchers, and these two handle `~".+"`

Reply:
Yeah I realized. I still prefer consistency :) Makes for easy maintenance.